packages/react-components/react-carousel/library/src/components/useEmblaCarousel.ts

```diff
@@ -216,8 +247,9 @@ export function useEmblaCarousel(
   );
 
   React.useEffect(() => {
+    // Scroll to controlled values on update
     const currentActiveIndex = emblaApi.current?.selectedScrollSnap() ?? 0;
-
+    emblaOptions.current.startIndex = activeIndex;
```

Comment:
The same is done on 262, no?

Reply:
This one should cover controlled indexes while next is a full reInit.

Reply:
Oops, I see. Should we move it inside `if` condition below?

Reply:
I think it makes sense to have these ones segmented as reinit logic should never fire on index change (only enforce retaining the existing index as ref), while we can keep this one specifically local to an index update whether controlled or internal.